meson.build

```diff
@@ -142,6 +140,46 @@ if git.found()
 endif
 add_project_arguments('-DSWAY_VERSION=@0@'.format(version), language: 'c')
 
+# Compute the relative path used by compiler invocations.
+source_root = meson.source_root().split('/')
+build_root = meson.build_root().split('/')
+relative_dir_parts = []
+i = 0
+in_prefix = true
+foreach p : build_root
+	if i >= source_root.length() or not in_prefix or p != source_root[i]
+		in_prefix = false
+		relative_dir_parts += '..'
+	endif
+	i += 1
+endforeach
+i = 0
+in_prefix = true
+foreach p : source_root
+	if i >= build_root.length() or not in_prefix or build_root[i] != p
+		in_prefix = false
+		relative_dir_parts += p
+	endif
+	i += 1
+endforeach
+relative_dir = join_paths(relative_dir_parts) + '/'
```

Comment:
An alternative to this algorithm is to just call into the python executable instead:
```meson
py = import('python').find_installation('python3')
relpath_cmd = '''
from os.path import relpath
from sys import argv
print(relpath(argv[1], start=argv[2]))
'''

relpath = run_command(py, '-c', relpath_cmd, meson.current_source_dir(), meson.build_root()).stdout().strip()
```
It's not a new build-time dependency, because meson already needs it.

Reply:
-1